include/eve/module/real/proba/function/regular/generic/cauchy.hpp

```diff
@@ -0,0 +1,194 @@
+    }
+
+    cauchy(T m_,  T s_) : m(m_), s(s_){
+      EVE_ASSERT(all(is_gtz(s) && is_finite(s)), "s must be strictly positive and finite");
+      EVE_ASSERT(all(s_finite(m)), "m must be finite");
+    }
+
+    cauchy(T m_) : m(m_), s(T(1)){
+      EVE_ASSERT(all(s_finite(m)), "m must be finite");
+    }
+
+    cauchy() : m(T(0)), s(T(1))   {}
+
+    T m, s;
+  };
+
+
+  namespace detail
+  {
+   //////////////////////////////////////////////////////
+    /// cdf
+    template<floating_value T, floating_value U>
+    EVE_FORCEINLINE  auto cdf_(EVE_SUPPORTS(cpu_)
+                              , cauchy<T> const &expo
+                              , U const &x ) noexcept
+    requires compatible_values<T, U>
+    {
+      return arithmetic_call(cdf, expo, x);
+    }
+
+    template<floating_value T>
+    EVE_FORCEINLINE  auto cdf_(EVE_SUPPORTS(cpu_)
+                              , cauchy<T> const &expo
+                              , T const &x ) noexcept
+    {
+      return half(as(x)) + atanpi((x-expo.m)/expo.s);;
+    }
+
+    //////////////////////////////////////////////////////
+    /// pdf
+    template<floating_value T, floating_value U>
+    EVE_FORCEINLINE  auto pdf_(EVE_SUPPORTS(cpu_)
+                              , cauchy<T> const &expo
+                              , U const &x ) noexcept
+    requires compatible_values<T, U>
+    {
+      return arithmetic_call(pdf, expo, x);
+    }
+
+    template<floating_value T>
+    EVE_FORCEINLINE  auto pdf_(EVE_SUPPORTS(cpu_)
+                              , cauchy<T> const &expo
+                              , T const &x ) noexcept
+    {
+      auto xmm = (x-expo.m);
+      return expo.s/(pi(as(x))*(sqr(expo.s)+sqr(xmm)));
+}
+
+    //////////////////////////////////////////////////////
+    /// invcdf
+    template<floating_value T, floating_value U>
+    EVE_FORCEINLINE  auto invcdf_(EVE_SUPPORTS(cpu_)
+                                 , cauchy<T> const &expo
+                                 , U const &x ) noexcept
+    requires compatible_values<T, U>
+    {
+      return arithmetic_call(invcdf, expo, x);
+    }
+
+    template<floating_value T>
+    EVE_FORCEINLINE  auto invcdf_(EVE_SUPPORTS(cpu_)
+                                 , cauchy<T> const &expo
+                                 , T const &x ) noexcept
+    {
+      auto tmp = fma(tanpi(x-half(as(x))), expo.s, expo.m);
+      // as x is restricted to [0,  1] limits values at 0 and 1 are properly defined
+      tmp = if_else(is_eqz(x), minf(as(x)), tmp);
+      tmp = if_else(x == one(as(x)), inf(as(x)), tmp);
+      return if_else(is_ltz(x) || x > one(as(x)), allbits, tmp);
+    }
+
+    //////////////////////////////////////////////////////
+    /// parameters
+    template<floating_value T>
+    EVE_FORCEINLINE  auto parameters_(EVE_SUPPORTS(cpu_)
+                                     , cauchy<T> const &expo) noexcept
+    {
+      return std::make_tuple(expo.m,  expo.s);
```

Comment:
I think this make_tuple is unneeded as one can use structured binding to retrieve m and s properly